### ISSUE TITLE
Integers are cast to floats when using context.type = json

### ISSUE THREAD

**Issue by Chive** [NikolaLohinski/terraform-provider-jinja#4]
Hi there, thanks for this awesome provider 🐐 

We've been running into an issue where integers are being automatically cast into floats when using the `json` context type. I've modified your example in https://github.com/NikolaLohinski/terraform-provider-jinja/tree/main/examples to reproduce the issue:

### `template.j2`
```jinja
This recipe will take {{ minutes }} minutes

The ingredients are:
{%- for ingredient in recipe %}
- {{ ingredient }}
{%- endfor %}
```

### `schema.json`
```json
{
    "type": "object",
    "required": [
        "recipe",
        "minutes"
    ],
    "properties": {
        "recipe": {
            "type": "array",
            "items": {
              "type": "string"
            },
            "minItems": 1,
            "maxItems": 5
        },
        "minutes": {
            "type": "integer"
        }
    }
}
```

### `context.json`

```json
{
  "recipe": [
    "salt",
    "meat",
    "vegetables"
  ],
  "minutes": 60
}
```

### `template.tf`
```tf
data "jinja_template" "render" {
  // must be a path to resolve any nested templates includes
  template = "${path.module}/src/template.j2"
  context {
    // either yaml or json
    type = "json"
    data = "${path.module}/src/context.json"
  }
  // is a list of either a path or inline, or both
  schemas = ["${path.module}/src/schema.json"]

  strict_undefined = false
  header           = "some macro for example"
  footer           = <<-EOF
    some value
  EOF
}
```

## Expected Behaviour

```bash
$ tf plan
data.jinja_template.render: Reading...
data.jinja_template.render: Read complete after 0s [id=xxx]

Changes to Outputs:
  + result = <<-EOT
        some macro for example
        This recipe will take 60 minutes

        The ingredients are:
        - salt
        - meat
        - vegetables

        some value
    EOT

[...]
```

Please note, this is this is the same behaviour with `context.type = yaml`.

## Actual behaviour

```bash
$ tf plan
data.jinja_template.render: Reading...
data.jinja_template.render: Read complete after 0s [id=xxx]

Changes to Outputs:
  + result = <<-EOT
        some macro for example
        This recipe will take 60.0 minutes 

        The ingredients are:
        - salt
        - meat
        - vegetables

        some value
    EOT

[...]
```

**Comment by NikolaLohinski** [comment]
Fixed in v1.4.1

Let me know if you run into any issues, or feel free to close the issue :)

**Comment by Chive** [comment]
All good, thanks a lot @NikolaLohinski
